vike/shared/page-configs/Config.ts

```diff
@@ -200,12 +200,14 @@ type OnRenderHtmlSync = (pageContext: PageContextServer) =>
       pageContext: Partial<Vike.PageContext> | (() => Promise<Partial<Vike.PageContext>>)
     }
 /** @deprecated Use a sync route() with an async guard() instead */
-type RouteAsync = (pageContext: PageContextServer | PageContextClient) => Promise<{ routeParams: Record<string, string> }>
+type RouteAsync = (
+  pageContext: PageContextServer | PageContextClient
+) => Promise<{ routeParams: Record<string, string> } | false>
 /** The page's URL(s).
  *
  *  https://vike.dev/route
  */
-type RouteSync = (pageContext: PageContextServer | PageContextClient) => { routeParams: Record<string, string> }
+type RouteSync = (pageContext: PageContextServer | PageContextClient) => { routeParams: Record<string, string> } | false
```

Comment:
FYI I have just discovered in https://vike.dev/route that this hook can also return `false`

Reply:
👍 It can actually be a `boolean` and it also has a [`precendence` option](https://vike.dev/route-function#precedence).

Reply:
I just double checked and, unless I forgot something, the return types of the hook types we added in your previous PR look good.

Reply:
> It can actually be a `boolean`

Out of curiosity: what would returning `true` mean?

Reply:
Ah I guess it's like returning no `routeParams`, so like returning `{}`

Reply:
Yes exactly.

Reply:
> 👍 It can actually be a `boolean` and it also has a [`precendence` option](https://vike.dev/route-function#precedence).

fixed